internal/handlers/handler_firstfactor.go

```diff
@@ -72,6 +72,15 @@ func FirstFactorPOST(delayFunc middlewares.TimingAttackDelayFunc) middlewares.Re
 			return
 		}
 
+		domainSession, err := ctx.GetDomainSession()
+		if err != nil {
+			ctx.Logger.Errorf("%s", err)
+
+			respondUnauthorized(ctx, messageAuthenticationFailed)
+
+			return
+		}
```

Comment:
```suggestion
```

Shouldn't this be excluded and be part of GetSession()?

Reply:
yes and no, domainSession is needed in line 105
```go
keepMeLoggedIn := domainSession.RememberMe != schema.RememberMeDisabled && bodyJSON.KeepMeLoggedIn != nil && *bodyJSON.KeepMeLoggedIn
```
in this case i think the best is not to use `ctx.GetSession`, and use `domainSession.GetSession()`